Source/MediaInfo/Audio/File_Usac.cpp

```diff
@@ -1803,16 +2252,485 @@ void File_Usac::UsacFrame(size_t BitsNotIncluded)
+        for (;;)
+        {
+            int16u pki=(int16u)-1;
+            // arith_get_pk
+            {
+                size_t i_min=-1;
+                size_t i_max=sizeof(arith_lookup_m)/sizeof(int16u)-1;
+                size_t i=i_min;
+                size_t j;
+                while (i_max-i_min>1)
+                {
+                    i=i_min+((i_max-i_min)/2);
+                    j=arith_hash_m[i];
+                    if (state+(esc_nb<<17)<(j>>8))
+                        i_max=i;
+                    else if (state+(esc_nb<<17)>(j>>8))
+                        i_min=i;
+                    else
+                    {
+                        pki=j&0xFF;
+                        break;
+                    }
+                }
+                if (pki==(int16u)-1)
+                    pki=arith_lookup_m[i_max];
+            }
+
+            m=arith_decode(low, high, value, arith_cf_m[pki], sizeof(arith_cf_m[pki])/sizeof(int16u));
+
+            if (m<ARITH_ESCAPE)
+                break;
+
+            if ((esc_nb=++lev)>7)
+                esc_nb=7;
+
+            if (lev>23)
+                return; //TODO: Handle error
+        }
+
+        b=m>>2;
+        a=m-(b<<2);
+
+        if (m==0)
+        {
+            if (lev) // ARITH_STOP symbol detected
+                break;
+
+            C.arithContext[ch].q[1][offset] = ((a + b + 1) <= 0x000F ? (a + b + 1) : 0x000F); // arith_update_context
+            continue;
+        }
+
+        for (int32u l=lev; l>0; l--)
+        {
+            int8u lsbidx=(a==0)?1:((b==0)?0:2);
+            int32u r=arith_decode(low, high, value, arith_cf_r[lsbidx], sizeof(arith_cf_r[lsbidx])/sizeof(int16u));
+
+            a=(a<<1)|(r&1);
+            b=(b<<1)|((r>>1)&1);
+        }
+
+        x_ac_dec[2*offset]=a;
+        x_ac_dec[2*offset+1]=b;
+        C.arithContext[ch].q[1][offset] = ((a + b + 1) <= 0x000F ? (a + b + 1) : 0x000F); // arith_update_context
+    }
+
+     // arith_finish
+     if (offset<sizeof(C.arithContext[ch].q[1]))
+         memset(&C.arithContext[ch].q[1][offset], 1, sizeof(C.arithContext[ch].q[1])-offset);
+
+     //Hacky way to rewind by 14
+     auto BS_Diff=BS_Sav.Remain()-BS->Remain();
+     if (BS_Diff>=14)
+         BS_Diff-=14;
+     *BS=BS_Sav;
+     BS->Skip(BS_Diff);
+
+     for (int16u i=0; i<lg; i++)
+     {
+         if (x_ac_dec[i]!=0)
+         {
+             TESTELSE_SB_SKIP(                                   "s");
+             TESTELSE_SB_ELSE(                                   "s");
+                 x_ac_dec[i]*=-1;
+             TESTELSE_SB_END()
+         }
+     }
+
+     Element_End0();
+ }
+
+//---------------------------------------------------------------------------
+void File_Usac::acSpectralData(size_t ch, bool usacIndependencyFlag)
+{
+    Element_Begin1("ac_spectral_data");
+
+    bool arith_reset_flag=true;
+    if (!usacIndependencyFlag)
+        Get_SB (arith_reset_flag,                               "arith_reset_flag");

```

Comment:
Add:

```

    if (!Aac_sampling_frequency[C.sampling_frequency_index])
    {
        Element_End0();
        return;
    }
```

(for smooth stop of parsing in case of unknown frequency index)